src/options.js

```diff
@@ -9,12 +9,16 @@ const objectToString = Object.prototype.toString;
 /** @jsdoc valueof */
 export function valueof(data, value, type) {
   const valueType = typeof value;
+  const access =
+    type && Object.getPrototypeOf(type) === TypedArray
```

Comment:
This should handle subclasses of typed arrays, too. For example, given:

```js
TypedArray = Object.getPrototypeOf(Uint8Array)
class MyUint8Array extends Uint8Array {}
```

Then

```
MyUint8Array instanceof TypedArray // false
Uint8Array instanceof TypedArray // false
Object.getPrototypeOf(MyUint8Array) === TypedArray // false
Object.getPrototypeOf(Uint8Array) === TypedArray // true
```

The fix here is pretty simple, I think:

```suggestion
    isTypedArray(type?.prototype)
```

Reply:
makes sense, but do we need the function call? : `type?.prototype instanceof TypedArray`

Reply:
Short functions like this typically get inlined anyway so it shouldn’t matter. And I figured we already had the helper.